source/tests/integration/ni_fake_non_ivi_service_tests_endtoend.cpp

```diff
@@ -0,0 +1,81 @@
+#include <gtest/gtest.h>
+#include <nifake_non_ivi/nifake_non_ivi_mock_library.h>
+#include <nifake_non_ivi/nifake_non_ivi_service.h>
+#include <server/async_processing_loop.h>
+
+#include <memory>
+#include <thread>
+
+namespace ni {
+namespace tests {
+namespace integration {
+
+using namespace nifake_non_ivi_grpc;
+
+// Test NiFakeNonIviService using a client stub.
+class NiFakeNonIviServiceTests_EndToEnd : public ::testing::Test {
+ public:
+  using FakeResourceRepository = nidevice_grpc::SessionResourceRepository<FakeHandle>;
+  nidevice_grpc::SessionRepository session_repository_;
+  std::shared_ptr<FakeResourceRepository> resource_repository_;
+  ni::tests::unit::NiFakeNonIviMockLibrary library_;
+  NiFakeNonIviService service_;
+  std::unique_ptr<::grpc::ServerCompletionQueue> completion_queue_;
+  std::thread callback_thread_;
+  std::unique_ptr<::grpc::Server> server_;
+  std::unique_ptr<NiFakeNonIvi::Stub> stub_;
+
+  NiFakeNonIviServiceTests_EndToEnd()
+      : session_repository_(),
+        resource_repository_(std::make_shared<FakeResourceRepository>(&session_repository_)),
+        library_(),
+        service_(&library_, resource_repository_),
+        server_(start_server()),
+        stub_(NiFakeNonIvi::NewStub(server_->InProcessChannel(::grpc::ChannelArguments())))
+  {
+  }
+
+  virtual ~NiFakeNonIviServiceTests_EndToEnd()
+  {
+    server_->Shutdown();
+    completion_queue_->Shutdown();
+    callback_thread_.join();
+  }
+
+  std::unique_ptr<::grpc::Server> start_server()
+  {
+    ::grpc::ServerBuilder builder;
+    builder.RegisterService(&service_);
+    completion_queue_ = builder.AddCompletionQueue();
+    auto server = builder.BuildAndStart();
+    callback_thread_ = std::thread([&]() { nidevice_grpc::run_async_processing_loop(completion_queue_.get(), service_); });
+    return server;
+  }
+
+  std::unique_ptr<NiFakeNonIvi::Stub>& stub()
+  {
+    return stub_;
+  }
+};
+
+TEST_F(NiFakeNonIviServiceTests_EndToEnd, ReadStream_ReturnsExpectedSequence)
```

Comment:
TODO: Oy! This test passes locally but fails a grpc assert when running the workflow. I need to figure out what that's about. I think the other integration tests have the same dependencies (i.e., "I need to run an in process server; Don't need drivers"). But maybe there's something wrong with my setup code.